File: PROJECT_C++/PROJECT_C++/Owca.cpp
#include "Owca.h"
#include <iostream>
#include <string>
#include <vector>
#include "Swiat.h"

Owca::Owca(int x, int y, Swiat& swiat) : Zwierze(4, 4, x, y, swiat) {};

void Owca::rysowanie() {
	std::cout << "O";
}

void Owca::rozmnozSie(int x, int y, Swiat& swiat) {
	Owca* nowyOrganizm = new Owca(x, y, swiat);
	swiat.dodajOrganizm(*nowyOrganizm);
}

Owca::~Owca() {};

File: OOPJava/src/main/game/zwierzeta/Owca.java
package main.game.zwierzeta;
import main.game.*;

public class Owca extends Zwierze{
    public Owca(int x, int y, Swiat swiat){ super(4,4,x,y,swiat); }
    public Owca(int sila, int inicjatywa, int x, int y, int poprzednieX, int poprzednieY, int wiek, Swiat swiat) {
        super(sila, inicjatywa, x, y, poprzednieX, poprzednieY, wiek, swiat);
    };

    @Override
    public String rysowanie(){
        return "owca.png";
    }
    @Override
    public void rozmnozSie(int x, int y, Swiat swiat){
        Owca nowyOrganizm = new Owca(x, y, swiat);
        swiat.dodajOrganizm(nowyOrganizm);
    }

    @Override
    public String getImie() { return "Owca"; };
}
